front/components/assistant/conversation/input_bar/InputBarAttachmentsPicker.tsx

```diff
@@ -71,6 +88,81 @@ export const InputBarAttachmentsPicker = ({
     minLength: MIN_SEARCH_QUERY_SIZE,
   });
 
+  const handleToolNodeSelect = useCallback(
+    async (node: ToolSearchNode) => {
+      const nodeKey = `${node.serverViewId}-${node.internalId}`;
+
+      if (loadingToolNodes.has(nodeKey)) {
+        return;
+      }
+      const existingBlob = fileUploaderService.fileBlobs.find(
+        (blob) => blob.id === nodeKey
+      );
+      if (existingBlob) {
+        fileUploaderService.removeFile(nodeKey);
+        return;
+      }
+
+      setLoadingToolNodes((prev) => new Set(prev).add(nodeKey));
+
+      try {
+        const response = await fetch(
+          `/api/w/${owner.sId}/mcp/views/${encodeURIComponent(node.serverViewId)}/attach?fileId=${encodeURIComponent(node.internalId)}`
+        );
+        if (!response.ok) {
+          const error = await response.json();
+          throw new Error(error.error?.message ?? "Failed to fetch file");
+        }
+
+        const fileData: ToolAttachment = await response.json();
+
+        // TODO: This is inefficient - we fetch the file as base64 from the server, convert it
+        // to a File object here, then upload it back to our file storage. Ideally the /attach
+        // endpoint should upload directly to file storage and return the file ID.
+        const binaryString = atob(fileData.contentBase64);
+        const bytes = new Uint8Array(binaryString.length);
+        for (let i = 0; i < binaryString.length; i++) {
+          bytes[i] = binaryString.charCodeAt(i);
+        }
+
+        if (!isSupportedFileContentType(fileData.mimeType)) {
+          throw new Error(`Unsupported file type: ${fileData.mimeType}`);
+        }
+
+        const blob = new Blob([bytes], { type: fileData.mimeType });
+        const file = new File([blob], fileData.fileName, {
+          type: fileData.mimeType,
+        });
+        await fileUploaderService.handleFilesUpload([file]);
+        setIsOpen(false);
+      } catch (error) {
```

Comment:
I think we should not wrap the whole thing in try/catch. 
Only where we "expect" there can be an error (so the fetch + the !respons.ok)